### DISCUSSION TITLE
Will the query results be cached?

### DISCUSSION THREAD

**Discussion by fujidaiti** [isar/isar#1232] [Help] [answered]
I have a question about the performance of Isar. If I implement pagination logic using `limit` and `offset`, and split the query result into N pages, will Isar execute the query N times in the database? Or does it execute it only once and cache the result for subsequent queries?

**Comment by Jtplouffe** [comment, marked as answer]
Isar does not do query caching. For every new query, the internal cursor will need to reach the desired offset again.
However, Isar is very fast, and loading times should be negligible, even if there is a couple thousands of entries.
